reactioncommerce/reaction
# Revision control for images

Resolves: #1456 
As it says, adds revision control to images. As with the other revision the main cruz of what is happening in `hooks` where the changes are deferred and moved into a revision record.

I made some schema changes to accommodate changes coming from other than the Products collection.

The biggest challenge was that the `Media` collection does not act like other collections because it is not a regular collection. Specifically it does not seem to allow attributes to be assigned (e.g. adding the `__revisions` property like is done for Products`) so assigning this had to happen on th client-side.

How to test:
Add/remove/edit images. You should not see these changes published until you hit "Publish Changes". When there is a pending change, you should see the "edit" icon on an image. (the only thing you can really "edit" about an image is the sort order)
